components/raftstore/src/store/fsm/store.rs

```diff
@@ -1752,6 +1752,23 @@ impl<'a, EK: KvEngine, ER: RaftEngine, T: Transport> StoreFsmDelegate<'a, EK, ER
             MEMTRACE_RAFT_MESSAGES.trace(TraceEvent::Sub(heap_size));
         });
 
+        if !self
+            .ctx
+            .global_replication_state
+            .lock()
```

Comment:
I'm afraid this can be the bottleneck of the system since raft messages are very frequent. I would be better to implement it on the network side. If it's in blocklist, the connection should be aborted and rejected on established.

Reply:
That way could work too. But I have not found a way to notify to network layer (Transport, RaftClient, etc.) as soon as the replication state is changed. To be more rigorous, we would actually also need the network layer to return success when the setup is complete before the lower layers can continue to synchronize new state to the PD. Do you have any suggestions for implementing such a notification mechanism?